patches/node/fix_-wunused-but-set-variable_warning.patch

```diff
@@ -1,323 +0,0 @@
-+#endif
- 
-   /* If the watcher wasn't started, this is a no-op. */
-   if (handle->signum == 0)
-@@ -166,8 +168,12 @@ int uv_signal_stop(uv_signal_t* handle) {
- 
-   EnterCriticalSection(&uv__signal_lock);
- 
-+#ifndef NDEBUG
-   removed_handle = RB_REMOVE(uv_signal_tree_s, &uv__signal_tree, handle);
-   assert(removed_handle == handle);
-+#else
-+  RB_REMOVE(uv_signal_tree_s, &uv__signal_tree, handle);
-+#endif
- 
-   LeaveCriticalSection(&uv__signal_lock);
- 
-diff --git a/deps/uv/src/win/thread.c b/deps/uv/src/win/thread.c
-index 89c53ada755b175f090faf9a00abd943140581c1..76db616047fa6d37e76dfe05c7f3d988cc45f39c 100644
---- a/deps/uv/src/win/thread.c
-+++ b/deps/uv/src/win/thread.c
-@@ -33,7 +33,9 @@
- #include "internal.h"
- 
- static void uv__once_inner(uv_once_t* guard, void (*callback)(void)) {
-+#ifndef NDEBUG
-   DWORD result;
-+#endif
-   HANDLE existing_event, created_event;
- 
-   created_event = CreateEvent(NULL, 1, 0, NULL);
-@@ -50,16 +52,24 @@ static void uv__once_inner(uv_once_t* guard, void (*callback)(void)) {
-     /* We won the race */
-     callback();
- 
-+#ifndef NDEBUG
-     result = SetEvent(created_event);
-     assert(result);
-+#else
-+    SetEvent(created_event);
-+#endif
-     guard->ran = 1;
- 
-   } else {
-     /* We lost the race. Destroy the event we created and wait for the existing
-      * one to become signaled. */
-     CloseHandle(created_event);
-+#ifndef NDEBUG
-     result = WaitForSingleObject(existing_event, INFINITE);
-     assert(result == WAIT_OBJECT_0);
-+#else
-+    WaitForSingleObject(existing_event, INFINITE);
-+#endif
-   }
- }
- 
-diff --git a/deps/uv/src/win/util.c b/deps/uv/src/win/util.c
-index 5ffde08e1aed041c4da679156ed10f7e54bfc386..056dd1e28ea68f5c363fa36dac175743d6cd0ccb 100644
---- a/deps/uv/src/win/util.c
-+++ b/deps/uv/src/win/util.c
-@@ -692,6 +692,7 @@ int uv_cpu_info(uv_cpu_info_t** cpu_infos_ptr, int* cpu_count_ptr) {
-     DWORD cpu_speed_size = sizeof(cpu_speed);
-     WCHAR cpu_brand[256];
-     DWORD cpu_brand_size = sizeof(cpu_brand);
-+#ifndef NDEBUG
-     size_t len;
- 
-     len = _snwprintf(key_name,
-@@ -700,6 +701,12 @@ int uv_cpu_info(uv_cpu_info_t** cpu_infos_ptr, int* cpu_count_ptr) {
-                      i);
- 
-     assert(len > 0 && len < ARRAY_SIZE(key_name));
-+#else
-+    _snwprintf(key_name,
-+               ARRAY_SIZE(key_name),
-+               L"HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\%d",
-+               i);
-+#endif
- 
-     err = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
-                         key_name,
-diff --git a/src/crypto/crypto_common.cc b/src/crypto/crypto_common.cc
-index 9cbe78c05b1c86c0d4a4b8b1f8771862dd3002e5..023ebc2e4cb343cb30fcab3cfc81b9cd6b200cd1 100644
---- a/src/crypto/crypto_common.cc
-+++ b/src/crypto/crypto_common.cc
-@@ -778,14 +778,12 @@ MaybeLocal<Array> GetClientHelloCiphers(
-     Environment* env,
-     const SSLPointer& ssl) {
-   EscapableHandleScope scope(env->isolate());
--  const unsigned char* buf = nullptr;
```

Comment:
Thanks for addressing! `src/crypto/crypto_common.cc` is valid for upstream as `buf` is not used anywhere.

Reply:
@deepak1556 we need to leave this but we can fold it into another patch - it's used in `SSL_CIPHER_find` and `SSL_client_hello_get0_ciphers`, but we comment those out for BoringSSL compat.